# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaManager.cpp
# Commit: 20543d03a4cd
# Full Hash: 20543d03a4cda05c785011a25079aa42aec8b2ba
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-03-29 21:53:01
# Regressor Bug: 1494675
# File Overlap Count: 2
# Description:
#   Bug 1494675 - Remove AllocationHandle. r=padenot
#   
#   The handle was used to keep separate allocations of the same source in a single
#   process apart. Sources no longer use sharing, so we no longer need allocations
#   or their handles even as a concept.
# ==============================================================================

diff -r 9e7832fee028 -r 20543d03a4cd dom/media/MediaManager.cpp
--- a/dom/media/MediaManager.cpp	Wed Mar 27 14:10:27 2019 +0000
+++ b/dom/media/MediaManager.cpp	Wed Mar 27 14:10:24 2019 +0000
@@ -6,7 +6,6 @@
 
 #include "MediaManager.h"
 
-#include "AllocationHandle.h"
 #include "AudioDeviceInfo.h"
 #include "MediaStreamGraphImpl.h"
 #include "MediaTimer.h"
@@ -915,7 +914,6 @@
   MOZ_ASSERT(MediaManager::IsInMediaThread());
   MOZ_ASSERT(mSource);
   return mSource->Allocate(aConstraints, aPrefs, mID, aPrincipalInfo,
-                           getter_AddRefs(mAllocationHandle),
                            aOutBadConstraint);
 }
 
@@ -924,13 +922,13 @@
                            const PrincipalHandle& aPrincipalHandle) {
   MOZ_ASSERT(MediaManager::IsInMediaThread());
   MOZ_ASSERT(mSource);
-  mSource->SetTrack(mAllocationHandle, aStream, aTrackID, aPrincipalHandle);
+  mSource->SetTrack(aStream, aTrackID, aPrincipalHandle);
 }
 
 nsresult MediaDevice::Start() {
   MOZ_ASSERT(MediaManager::IsInMediaThread());
   MOZ_ASSERT(mSource);
-  return mSource->Start(mAllocationHandle);
+  return mSource->Start();
 }
 
 nsresult MediaDevice::Reconfigure(
@@ -938,26 +936,25 @@
     const MediaEnginePrefs& aPrefs, const char** aOutBadConstraint) {
   MOZ_ASSERT(MediaManager::IsInMediaThread());
   MOZ_ASSERT(mSource);
-  return mSource->Reconfigure(mAllocationHandle, aConstraints, aPrefs, mID,
-                              aOutBadConstraint);
+  return mSource->Reconfigure(aConstraints, aPrefs, mID, aOutBadConstraint);
 }
 
 nsresult MediaDevice::FocusOnSelectedSource() {
   MOZ_ASSERT(MediaManager::IsInMediaThread());
   MOZ_ASSERT(mSource);
-  return mSource->FocusOnSelectedSource(mAllocationHandle);
+  return mSource->FocusOnSelectedSource();
 }
 
 nsresult MediaDevice::Stop() {
   MOZ_ASSERT(MediaManager::IsInMediaThread());
   MOZ_ASSERT(mSource);
-  return mSource->Stop(mAllocationHandle);
+  return mSource->Stop();
 }
 
 nsresult MediaDevice::Deallocate() {
   MOZ_ASSERT(MediaManager::IsInMediaThread());
   MOZ_ASSERT(mSource);
-  return mSource->Deallocate(mAllocationHandle);
+  return mSource->Deallocate();
 }
 
 dom::MediaSourceEnum MediaDevice::GetMediaSource() const {